### PR TITLE
automatic email verification

### LINKED ISSUE #9710: Changing email does not automatically resend verification email
### Description:

Changing my email address should automatically send me a a verification email. As a result, I need to click the 'resend verification email' button. There is currently a bug that makes that resend button not work: for this see #7577 

resend

### Server Setup Information:

* Version of Rocket.Chat Server: 0.59

### Steps to Reproduce:

1. Change email address
2. Button occurs

### Expected behavior:

The email should be sent automatically without user interaction.

### Actual behavior:

The user has to click the resend button.

### PR DESCRIPTION
**Pull request by abhishek71994**
[NEW] resolve #9710 

Closes #9710

### CODE REVIEW

File: packages/rocketchat-ui-account/client/accountProfile.html [hunk 1]
```diff
@@ -130,11 +130,6 @@
 								{{#unless canChange}}
 									<div class="rc-input__description">{{_ 'Email_Change_Disabled'}}</div>
 								{{/unless}}
-								{{#unless emailVerified}}
-									<div class="rc-input__error">
-										<button class="rc-button rc-button--small" id="resend-verification-email">{{_ "Resend_verification_email"}}</button>
-									</div>
-								{{/unless}}
```
[the remark this answers is not in the document]
Comment (abhishek71994) [reply packages/rocketchat-ui-account/client/accountProfile.html]: From the feedback. I was under the impression that the feature update that was required needed the button to be removed and the user update to happen only when the user clicked the "Save changes" button. Please let me know How would you suggest I go about with this? That would be really helpful.

[the remark this answers is not in the document]
Comment (geekgonecrazy) [reply packages/rocketchat-ui-account/client/accountProfile.html]: The issue isn't about the button at all.

If the users email isn't verified this button should show up regardless.

What should happen is when the user changes the email it should at that point trigger the verify email at the same time.

[the remark this answers is not in the document]
Comment (abhishek71994) [reply packages/rocketchat-ui-account/client/accountProfile.html]: So as the user types in the email the email verification event should begin?

**Comment by geekgonecrazy** [comment]
Let me add this top level so it doesn't get lost.  :)

The email verification button needs to change and always show if email is not verified.  Exactly as is.

What needs to be added is: When the user changes the email and hits save.  At that time it should trigger the email verification action.

This way when you change your email you receive an email right away.  And don't get locked out of your account later because you forgot to click the button.

**Comment by abhishek71994** [comment]
@geekgonecrazy I've added some changes. But I had a doubt. 
Suppose the user's old email is "xyz@gmail.com" and he/she changes it to "abc@gmail.com"
I saw that the email is sent to "xyz@gmail.com" rather than "abc@gmail.com" in the console.

**Comment by geekgonecrazy** [comment]
Yes exactly the way you have it, it's gonna use the old email, because the new email hasn't even been saved.  Follow above feedback and you should be good to go

**Comment by abhishek71994** [comment]
@geekgonecrazy any updates?

**Comment by geekgonecrazy** [comment]
It's still not yet right.  Need to save the profile before you send email.  

With a fresh look... I think there might be a safer approach

Maybe the best course of action would be to go-to the saveProfile function and do the send the email if the email is different instead of calling two functions from the client.  This way if the profile is updated from API this same thing will happen.

So no changes client side and call the confirmation method inside of that method.

@rodrigok thoughts?

**Comment by avimanyu786** [comment]
This still isn't working. I followed https://rocket.chat/docs/administrator-guides/email/setup/ and set it up with [sendgrid](sendgrid.com). SMTP Test Email is working.

**Comment by rodrigok** [comment]
We apologize for not getting a review on this sooner. The issue mentioned was already solved by [this commit](https://github.com/RocketChat/Rocket.Chat/commit/028f08aec483cab758587da08ac76e961e0c08b5)

**Pull request closed without merging**